src/specFunctions.py

```diff
@@ -208,11 +208,17 @@ def peakIndexes(y, thres=0.3, min_dist=1, thres_abs=False):
 
         # for each chain of zero indexes
         for plateau in zero_plateaus:
-            median = np.median(plateau)
-            # set leftmost values to leftmost non zero values
-            dy[plateau[plateau < median]] = dy[plateau[0] - 1]
-            # set rightmost and middle values to rightmost non zero values
-            dy[plateau[plateau >= median]] = dy[plateau[-1] + 1]
+            lplat = plateau.shape[0]
+            left = dy[plateau[0] - 1]
+            right = dy[plateau[lplat-1] + 1]
+            middle = lplat // 2
+            # always in sorted order
+            median = plateau[middle]
```

Comment:
It's actually faster to just loop over the `plateau` elements once and do something with each element than to do the vectorized ops they were performing before. The changes in this function result in a roughly 4+x speedup which is awesome because the peak finding logic is a big cause of slowness when running the loop.